examples/other/use-custom-fetch-composable/composables/useCustomFetch.ts

```diff
@@ -0,0 +1,34 @@
+import type { UseFetchOptions } from 'nuxt/app'
+
+export async function useCustomFetch<T> (url: string, options: UseFetchOptions<T> = {}) {
+  const userAuth = useCookie('token')
+  const config = useRuntimeConfig()
+
+  const defaults: UseFetchOptions<T> = {
+    baseURL: config.baseUrl ?? 'https://api.nuxtjs.dev',
+    // cache request
+    key: url,
+
+    onRequest ({ options }) {
+      options.headers = options.headers ?? new Headers()
+
+      // set user token if connected
+      if (userAuth.value) {
+        options.headers.Authorization = `Bearer ${userAuth.value}`
+      }
```

Comment:
we might also need to handle the case where this is a `Headers` object and we need to do `headers.set()`. For simplicity, is there any reason not to use the `headers: {}` option directly instead of within `onRequest`?

Reply:
tried something simpler :)